### PR TITLE
Migrate qmsi and esp32 wdt driver api to new generic wdt api interfaces. 

### PR DESCRIPTION
**Pull request by SavinayDharmappa**

### CODE REVIEW

**Comment by nashif** [comment]
getting this on esp32:

```
***** Booting Zephyr OS v1.13.0-rc3-11-g7e95ac313d *****
Watchdog sample application
Feeding watchdog 5 times
Feeding watchdog...
 ** FATAL EXCEPTION
 ** CPU 0 EXCCAUSE 20 PS 0x00060e20 PC 0x00000000 VADDR 0x00000000
 **  A0 0x80082081  SP 0x3ffb17b0  A2 0x3ffb0000  A3 0x00000004
 **  A4 0x00000000  A5 0x00060c20  A6 0x3ffb0fe8  A7 0x00000028
 **  A8 0x80080811  A9 0x3ffb1760 A10 0x3ffb0000 A11 0x00000000
 ** A12 0x3ffb01f1 A13 0x00000004 A14 0x00000000 A15 0x00000000
 ** LBEG 0x00000000 LEND 0x00000000 LCOUNT 0x00000000
 ** SAR 0x00000018
Current thread ID = 0x3ffb0b00
Faulting instruction address = 0xdeaddead
Fatal fault in ISR! Spinning...
```

File: drivers/watchdog/wdt_esp32.c [hunk 1]
```diff
@@ -64,44 +65,11 @@ static int wdt_esp32_disable(struct device *dev)
...
-	 *
-	 * While this is sort of backwards (this should be given units of
-	 * time and converted to what the hardware expects), try to map this
-	 * value to what the ESP32 expects.  Use the same timeout value for
-	 * stages 0 and 1, regardless of the configuration mode, in order to
-	 * simplify things.
-	 */
+	u32_t ticks = timeout;
 
-	 /* MWDT ticks every 12.5ns.  Set the prescaler to 40000, so the
```
Comment (nashif) [inline drivers/watchdog/wdt_esp32.c]: why are you removing those comments?

Comment (SavinayDharmappa) [reply drivers/watchdog/wdt_esp32.c]: ok. I will change it.

**Comment by SavinayDharmappa** [comment]
@galak I can do it for quark, But for esp32 still basic dts support needs to be addded. #7581

**Comment by nashif** [comment]
@SavinayDharmappa can you please rebase? and please do the DTS support for quark.

**Comment by SavinayDharmappa** [comment]
added device tree support for quark

**Review by nashif — APPROVED** [review]

**Pull request merged**